enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/analytics/AnalyticsFilter.java

```diff
@@ -74,6 +74,15 @@ private AnalyticsFilter() {
             // We are always expecting <String, String> Map as configuration.
             publisherConfig.put(entry.getKey(), getEnvValue(entry.getValue()).toString());
         }
+
```

Comment:
Do we initialize analytics filter only when analytics is enabled?

Reply:
Yes, as I have checked config is checked before getting a instance of AnalyticsFilter in all the places